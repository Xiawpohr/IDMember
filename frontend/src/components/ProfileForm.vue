<template>
  <form @submit.prevent="$emit('submitted', modifiedUser)">
    <p>profile form</p>
    <input type="text" v-model="firstName"/>
    <input type="text" v-model="lastName"/>
    <button type="submit">Save</button>
  </form>
</template>

<script>
export default {
  props: {
    user: {
      types: Object,
      default: () => ({
        firstName: '',
        lastName: ''
      })
    }
  },
  data() {
    return {
      firstName: this.user.firstName,
      lastName: this.user.lastName
    }
  },
  computed: {
    modifiedUser() {
      return {
        firstName: this.firstName,
        lastName: this.lastName
      }
    }
  }
}
</script>
