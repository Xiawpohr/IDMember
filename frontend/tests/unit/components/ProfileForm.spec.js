import { shallowMount } from '@vue/test-utils'
import ProfileForm from '@/components/ProfileForm.vue'

const user = { firstName: 'Arthur', lastName: 'Hsiao' }

let wrapper

beforeEach(() => {
  wrapper = shallowMount(ProfileForm)
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should call submitted event when submitting form', () => {
  wrapper.setData({
    ...user
  })
  const button = wrapper.find('button')
  button.trigger('click')
  button.trigger('submit')
  expect(wrapper.emitted('submitted')[0]).toEqual([user])
})
