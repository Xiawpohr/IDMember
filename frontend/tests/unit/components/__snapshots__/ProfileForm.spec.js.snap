// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<form>
  <p>profile form</p> <input type="text"> <input type="text"> <button type="submit">Save</button>
</form>
`;
